Toolset/palettes/script editor/behaviors/revsehandlerfilterbehavior.livecodescript

```diff
@@ -0,0 +1,90 @@
+﻿script "revSEHandlerFilterBehavior"
+
+/*
+The aim here is to display the text "Filter..." in italics when the
+field is not focused and there is no current filter text, and to
+display the filter text in normal font otherwise.
+
+When the content is modified, a message is sent immediately.
+The following custom properties are used to configure how the
+message is sent:
+
+    cCallback       -- the message to send
+    cCallbackTarget -- the long ID of the object that the message
+                       should be sent to
+*/
+
+----------------------------------------------------------------
+-- Helper functions for providing the placeholder text
+----------------------------------------------------------------
+
+private command startFiltering
+   if not the cFiltering of me then
+      put empty into me
+      set the cFiltering of me to true
+      send "update" to me
+   end if
+   selectAll
+end startFiltering
+
+private command stopFiltering
+   if me is empty then
+      set the cFiltering of me to false
+      send "update" to me
+   end if
+end stopFiltering
+
+private command selectAll
+   select char 1 to -1 of me
+end selectAll
+
+-- This does the main work of fixing up the formatting.  It also
+-- sends the registered callback message if necessary.
+on update
+   if the cFiltering of me then
+      set the textstyle of me to "normal"
```

Comment:
I don't think "normal" is a valid textStyle (although I believe invalid text styles default to empty anyway). Either "plain" or empty gives unstyled text.